src/keepalive/service/net/Client.java

```diff
@@ -31,7 +31,7 @@ public static FetchResult fetch(FreenetURI uri, HighLevelSimpleClientImpl hlsc)
         FetchContext fetchContext = hlsc.getFetchContext();
         fetchContext.returnZIPManifests = true;
         FetchWaiter fetchWaiter = new FetchWaiter(rc);
-        hlsc.fetch(uri, -1, fetchWaiter, fetchContext);
+        hlsc.fetch(uri, Long.MAX_VALUE, fetchWaiter, fetchContext);
```

Comment:
This should not work: it would expect the max value to be the lowest possible number.